src/key-value.spec.js

```diff
@@ -50,8 +53,12 @@ describe('ReactKeyValue', () => {
     ]);
   });
   it('should handle removing a row', () => {
-    const $ = shallow(<KeyValue rows={ [{ keyItem: 'a', valueItem: 'A' }] }/>);
-    $.instance().handleRemove(0);
+    const $ = shallow(
+      <KeyValue
+        rows={ [{ keyItem: 'a', valueItem: 'A' }] }
+      />
+    );
+    $.find('.key-value-row-remove').find('button').simulate('click');
     $.state('rows').should.eql([]);
```

Comment:
Technically, it's a great test, but I'd have some comments here, just for the record.
Here, we know that this code probably removes the proper element from the state, but can we be sure that it doesn't remove every element from the state? Or it might just removes always the first element, or the last element. All of the aforementioned cases could be true.
To improve this test, I'd provide an array of different cases (which includes the input and the expected output) and iterate through them just to make 100% sure that this function works in the expected way.